scripts/linux/smokeTestHelper.sh

```diff
@@ -90,6 +144,76 @@ check-github-pmc-artifacts-similarity()
   fi
 }
 
+check-pmc-images-availability()
+{
+  # Check docker images availability in the PMC against all image tags.
+  #
+  # $1 - Github branch name
+
+  sourceBranchName=$1
+  taskDisplayNames=("Publish Edge Agent Manifest" "Publish Edge Hub Manifest" "Publish Temperature Sensor Manifest" "Publish Diagnostic Module Manifest")
+  for taskDisplayName in "${taskDisplayNames[@]}"
+  do
+    # Amalgam of test cases to test docker images in mcr
+    echo $'\n\n================================================\n\n'
+    check-pmc-image-tags-availability "$sourceBranchName" "$taskDisplayName"
+  done
+}
+
+check-pmc-image-tags-availability()
+{
+  # Check if the images pulled using docker image name from MCR resolves the to the same image SHA 
+  # which "Azure-IoT-Edge-Core Images Publish" pipeline claimed to publish to the MCR. 
+  #
+  # $1 - branch name
+  # $2 - task display name (use to reference the image module)
+  
+  branchName=$1
+  pipelineDisplayName=$2
+  echo "Checking images published for $branchName ($pipelineDisplayName)"
+
+  buildId=$(get-latest-image-publication-run "$branchName")
+  echo "BuildId: $buildId"
+
+  logs=$(get-build-logs-from-task "$buildId" "$pipelineDisplayName")
+
+  imageHashMap=$(get-image-sha-from-devops-logs "$logs")
+  echo "Checking the following images publication:"
+  echo "$imageHashMap"
+  echo ""
+
+  OLD_IFS=$IFS; 
+  IFS=' '
+  nameList=($(echo "$imageHashMap" | awk '{sub(/.*azureiotedge/, "azureiotedge", $1); print $1}' | tr '\n' ' '))
+  shaList=($(echo "$imageHashMap" | awk '{print $2}' | tr '\n' ' '))
+  pmcImages=($(prepare-docker-image-names "$imageHashMap" | tr '\n' ' '))
+  
+  IFS=''
+  pullResults=()
+  for image in "${pmcImages[@]}"
+  do
+    pullResults+=($(docker pull $image))
+  done
+
+  IFS=$OLD_IFS
+  isFailed=false
+  for i in $(seq 0 $((${#nameList[@]} - 1 )) );
+  do
+    echo "${pullResults[$i]}" | grep -q "${nameList[$i]}" \
+      && { echo "[PASS] ${nameList[$i]} is available."; } \
+      || { echo "[FAIL] The image name:tag ( ${nameList[$i]} ) is missing."; isFailed=true; }
+    echo "${pullResults[$i]}" | grep -q "${shaList[$i]}" \
+      && { echo "[PASS] ${shaList[$i]} is available."; } \
+      || { echo "[FAIL] The sha256 ( ${shaList[$i]} ) is missing."; isFailed=true; }
+  done
```

Comment:
Can this all be done in one loop rather than 2?

Reply:
I'd rather have it pull all the images before checking them one at a time.